src/functions/module.php

```diff
@@ -2,7 +2,41 @@
 
 declare(strict_types=1);
 
+function drupal_alter(string|array $type, &$data, &$context1 = null, &$context2 = null, &$context3 = null)
+{
+    \Drupal::moduleHandler()->alter($type, $data, $context1, $context2);
+}
+
 function module_exists(string $module): bool
 {
     return \Drupal::moduleHandler()->moduleExists($module);
 }
+
+function module_implements(string $hook, ?bool $sort = false, ?bool $reset = false): array
+{
+    static $drupal_static_fast;
+    if (!isset($drupal_static_fast)) {
+        $drupal_static_fast['implementors'] = &drupal_static(__FUNCTION__, []);
+        $drupal_static_fast['sorted'] = &drupal_static(__FUNCTION__ . ':sorted', []);
+    }
+    $implementors = &$drupal_static_fast['implementors'];
+    if ($reset) {
+        $implementors  = [];
+        $sorted = [];
+        \Drupal::moduleHandler()->resetImplementations();
+    }
+    if (!isset($implementors[$hook])) {
+        $implementors[$hook] = [];
+        \Drupal::moduleHandler()->invokeAllWith($hook, function (callable $callback, string $module) use (&$implementors, $hook) {
+            $implementors[$hook][] = $module;
+        });
+    }
+    if ($sort) {
+        if (!isset($sorted[$hook])) {
+            $sorted[$hook] = $implementors[$hook];
+            sort($sorted[$hook]);
+        }
+        return $sorted[$hook];
+    }
+    return $implementors[$hook];
+}
```

Comment:
I'd rather decorate and override `ModuleHandler` to replace `hasImplementations` which proxies to protected `getImplementationInfo`

Reply:
I don't know how to override in a library yet. Not sure I would want to change the return type of `hasImplementations`. `ModuleHandler` used to provide the `getImplementations` method to return a list of modules. This code just follows the recommendation in the [change record](https://www.drupal.org/node/3000490) when that method was removed.